xarray/backends/zarr.py

```diff
@@ -0,0 +1,554 @@
+        # compressor='default', fill_value=0, order='C', store=None,
+        # synchronizer=None, overwrite=False, path=None, chunk_store=None,
+        # filters=None, cache_metadata=True, **kwargs)
+        print('creating', name, shape, dtype)
+        zarr_array = self.ds.create(name, shape=shape, dtype=dtype,
+                                    fill_value=fill_value, **encoding)
+        # decided not to explicity enumerate encoding options because we
+        # risk overriding zarr's defaults (e.g. if we specificy
+        # cache_metadata=None instead of True). Alternative is to have lots of
+        # logic in _extract_zarr_variable encoding to duplicate zarr defaults.
+        #                            chunks=encoding.get('chunks'),
+        #                            compressor=encoding.get('compressor'),
+        #                            filters=encodings.get('filters'),
+        #                            cache_metadata=encoding.get('cache_metadata'))
+
+        # the magic for storing the hidden dimension data
+        zarr_array.attrs[_DIMENSION_KEY] = dims
+        _, attributes = _get_zarr_dims_and_attrs(zarr_array, _DIMENSION_KEY)
+
+        for k, v in iteritems(attrs):
+            attributes[k] = _encode_zarr_attr_value(v)
+
+        return zarr_array, variable.data
+
+
+    def store(self, variables, attributes, *args, **kwargs):
+        new_vars = OrderedDict((k, encode_zarr_variable(v, name=k))
+                               for k, v in iteritems(variables))
+        AbstractWritableDataStore.store(self, new_vars, attributes,
+                                        *args, **kwargs)
+    # sync() and close() methods should not be needed with zarr
+
+
+# from zarr docs
+
+# Zarr arrays can be used as either the source or sink for data in parallel
+# computations. Both multi-threaded and multi-process parallelism are
+# supported. The Python global interpreter lock (GIL) is released for both
+# compression and decompression operations, so Zarr will not block other Python
+# threads from running.
+#
+# A Zarr array can be read concurrently by multiple threads or processes. No
+# synchronization (i.e., locking) is required for concurrent reads.
+#
+# A Zarr array can also be written to concurrently by multiple threads or
+# processes. Some synchronization may be required, depending on the way the
+# data is being written.
+
+# If each worker in a parallel computation is writing to a separate region of
+# the array, and if region boundaries are perfectly aligned with chunk
+# boundaries, then no synchronization is required. However, if region and chunk
+# boundaries are not perfectly aligned, then synchronization is required to
+# avoid two workers attempting to modify the same chunk at the same time.
+
+
+def open_zarr(store, mode='r+', group=None, synchronizer=None, auto_chunk=True,
+              decode_cf=True, mask_and_scale=True, decode_times=True,
```

Comment:
As long this remains experimental, can we drop all the decoding options or at least the other options that we haven't implement yet like cache/autoclose? We are probably going to need to refactor coding related options soon, though I guess there are worst things than a deprecation cycle.

Reply:
I'll remove cache and autoclose for now. encoding options are just passed through to decode_cf and should work. (They are covered by the tests via inheritance from `CFEncodedDataTest`.)

Reply:
Unless we actually implement `cache`, I think it's better to remove it from the argument list (at least prior to the public release). Otherwise it gives the misleading impression that the method is actually implemented.